### ISSUE TITLE
404 links in documentation when viewed with mdbook

### ISSUE THREAD

**Issue by 4ydan** [ElektraInitiative/PermaplanT#506]
## Steps to Reproduce the Problem

```sh
cargo install mdbook mdbook-mermaid
cargo install --git https://github.com/ElektraInitiative/mdbook-generate-summary mdbook-generate-summary
mdbook serve --open
```
* Navigate to http://localhost:3000/architecture/12glossary.html
* scroll down
* press on any of the two hyperlink "here"
![image](https://github.com/ElektraInitiative/PermaplanT/assets/53004139/aedf66cf-fb21-4ff4-911c-62eed8b79f41)

## Expected Result

Navigate to the proper markdown

## Actual Result

![image](https://github.com/ElektraInitiative/PermaplanT/assets/53004139/59244d78-de9a-46df-8154-66d35ec9e1d0)

## System Information

- PermaplanT: master
- Operating System: Inside a Rust Devcontainer in VSCode under Windows10
- Browser: Opera

## And
Basically all of these need to be fixed
![image](https://github.com/ElektraInitiative/PermaplanT/assets/53004139/9e383c23-ec9b-4d09-9ede-a34f7dfe3e07)

Weird to make it work offline in markdown and as mdbook, maybe replacing /doc/ with ../ might do the fix for most of these.

**Comment by markus2330** [comment]
Thank you, great findings! I can reproduce that these links are dead. backend was moved (now in doc/backend), the database/hierarchy.md seems to be some limitation of mdbook (seems like mdbook doesn't support absolute links).

Let us try out the link checker mdbook provides, maybe it would avoid such problems in future once we have it as part of CI.

**Comment by 4ydan** [comment]
What do we think about a pre-commit [hook](https://github.com/tcort/markdown-link-check#run-as-a-pre-commit-hook) solution to checking if the links work? Or is it maybe to inconsistent.

**Comment by markus2330** [comment]
If the mdbook link-checker solution is not working easily, it could be much better than nothing, depending on how similar the checks to what mdbook allows are (e.g. absolute links should be prohibited, as they don't work with mdbook afaik).

Imho broken links that lead to existing files not included in mdbook are not terribly problematic, people can open the file via some other way. (Of course the non-plus-ultra variant is if these links go to the repo, like implemented in libelektra.)
